torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -461,6 +465,31 @@ def __init__(
             if param not in ignored_params and not isinstance(param, FlatParameter):
                 params.append(param)
 
+        # At this point we are wrapping an atomic FSDP unit. Check module shapes
+        # and broadcast module from rank 0 to ensure everything starts off with
+        # the same parameters.
+        # Hack: move to compute device and back if it is CPU
+        prev_device = next(module.parameters()).device
+        was_cpu = (prev_device == torch.device("cpu"))
+        if was_cpu:
+            warnings.warn(
+                f"Module is input on CPU, we are moving to to {self.compute_device} to perform parameter verification and will move it back after. In the future please specify device_ids which will explicitly tell FSDP to place the module on a particular device."
```

Comment:
nit: `to to` -> `to`

`device_ids` -> `device_id`? (We want to enforce SPSD for FSDP?)

Reply:
Is it OK to mention `device_id` at this point, given that this arg is not available yet?

Reply:
Was planning to base the next PR under this one and land this after that one lands and we agree on CPU-side semantics. Currently PR is mostly for discussion

Reply:
Will remove it for now, and update warning when device_ids PR lands.